examples/builtin_transformations.rs

```diff
@@ -0,0 +1,147 @@
+use anyhow::Result;
+use compositor_common::{scene::Resolution, Framerate};
+use log::{error, info};
+use serde_json::json;
+use signal_hook::{consts, iterator::Signals};
+use std::{
+    process::{Command, Stdio},
+    thread,
+    time::Duration,
+};
+use video_compositor::http;
+
+use crate::common::write_example_sdp_file;
+
+#[path = "./common/common.rs"]
+mod common;
+
+const VIDEO_RESOLUTION: Resolution = Resolution {
+    width: 1920,
+    height: 1080,
+};
+
+const FRAMERATE: Framerate = Framerate(30);
+
+fn main() {
+    env_logger::init_from_env(
+        env_logger::Env::default().filter_or(env_logger::DEFAULT_FILTER_ENV, "info"),
+    );
+    ffmpeg_next::format::network::init();
+
+    thread::spawn(|| {
+        if let Err(err) = start_example_client_code() {
+            error!("{err}")
+        }
+    });
+
+    http::Server::new(8001).start();
+
+    let mut signals = Signals::new([consts::SIGINT]).unwrap();
+    signals.forever().next();
+}
+
+fn start_example_client_code() -> Result<()> {
+    thread::sleep(Duration::from_secs(2));
+
+    info!("[example] Sending init request.");
+    common::post(&json!({
+        "type": "init",
+        "framerate": FRAMERATE,
+        "init_web_renderer": false,
+    }))?;
+
+    info!("[example] Start listening on output port.");
+    let output_sdp = write_example_sdp_file("127.0.0.1", 8002)?;
+    Command::new("ffplay")
+        .args(["-protocol_whitelist", "file,rtp,udp", &output_sdp])
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .spawn()?;
```

Comment:
I feel like most of this code should be extracted somewhere at this point. It's ugly and, I think, very similar in every example

Reply:
Since those are examples, not tests, I would avoid putting abstraction here and keep it as simple as possible